src/client/application/diagnostics/checks/mplsSurvey.ts

```diff
@@ -0,0 +1,124 @@
+        @inject(IDisposableRegistry) disposableRegistry: IDisposableRegistry,
+        @inject(IApplicationEnvironment) private appEnvironment: IApplicationEnvironment,
+        @inject(IPlatformService) private platformService: IPlatformService,
+        @inject(IBrowserService) private browserService: IBrowserService,
+    ) {
+        super([DiagnosticCodes.MPLSSurveyDiagnostic], serviceContainer, disposableRegistry, true);
+    }
+
+    public async diagnose(resource: Resource): Promise<IDiagnostic[]> {
+        if (!this.shouldShowPrompt) {
+            return [];
+        }
+
+        return [new MPLSSurveyDiagnostic(ExtensionSurveyBanner.mplsMessage(), resource)];
+    }
+
+    protected async onHandle(diagnostics: IDiagnostic[]): Promise<void> {
+        if (diagnostics.length === 0 || !this.canHandle(diagnostics[0])) {
+            return;
+        }
+
+        const diagnostic = diagnostics[0];
+        if (await this.filterService.shouldIgnoreDiagnostic(diagnostic.code)) {
+            return;
+        }
+
+        await this.messageService.handle(diagnostic, {
+            commandPrompts: [
+                {
+                    prompt: ExtensionSurveyBanner.bannerLabelYes(),
+                    command: {
+                        diagnostic,
+                        invoke: () => this.launchSurvey(),
+                    },
+                },
+                {
+                    prompt: ExtensionSurveyBanner.maybeLater(),
+                    command: {
+                        diagnostic,
+                        invoke: async () => this.disable(),
+                    },
+                },
+                {
+                    prompt: ExtensionSurveyBanner.bannerLabelNo(),
+                    command: {
+                        diagnostic,
+                        invoke: () => this.updateMemento(),
+                    },
+                },
+            ],
+            onClose: () => this.disable(),
+        });
+    }
+
+    private async updateMemento() {
+        await this.context.globalState.update(MPLS_SURVEY_MEMENTO, true);
+    }
+
+    private disable() {
+        this.disabledInCurrentSession = true;
+    }
+
+    private get shouldShowPrompt(): boolean {
+        return (
+            env.uiKind !== UIKind?.Web &&
+            !this.disabledInCurrentSession &&
+            !this.context.globalState.get(MPLS_SURVEY_MEMENTO)
+        );
+    }
+
+    private async launchSurvey() {
+        const query = querystring.stringify({
+            o: encodeURIComponent(this.platformService.osType), // platform
+            v: encodeURIComponent(this.appEnvironment.vscodeVersion),
+            e: encodeURIComponent(this.appEnvironment.packageJson.version), // extension version
+            m: encodeURIComponent(this.appEnvironment.sessionId),
+        });
+        const url = `https://aka.ms/mpls-experience-survey?${query}`;
+        this.browserService.launch(url);
+        await this.updateMemento();
```

Comment:
Commands can be used instead, that way creating `MPLS_SURVEY_MEMENTO` is not required.
```suggestion
        const command = commandFactory.createCommand(diagnostic, { type: 'ignore', options: DiagnosticScope.Global });
        await command.invoke();
```

Reply:
The command system is completely confusing to me; what does "ignore" _mean_? Ignored in the session? Ignored for all time? Ignored for one specific workspace? Which one is it? I couldn't chase this around to figure it out.

Reply:
In here `{ type: 'ignore', options: DiagnosticScope.Global }`, the scope is `DiagnosticScope.Global` so it's ignored globally. (meaning `Do not show again`)

Reply:
I'll try it out, though I'll have to redo all of my tests to deal with the commands.